### PR TITLE
Email management API endpoints

### PR DESCRIPTION
**Pull request by chouseknecht**
The endpoints:
- View all emails, and create new email by posting to `/api/v1/emails`
- Update and Delete an emails at `/api/v1/emails/#/`
- View all emails for a user at `/api/v1/users/#/emails/`
- View user's primary email at `/api/v1/me/`

Security:
- An authenticated user can only update his/her own email addresses
- Non-admin can only view his/her own email addresses
- Anonymous users cannot view email addresses

Caveats:

There is no way to check that at least one email address is marked as primary, as I'm not sure where we would put that check. Wondering if we can check for a primary_email at `/api/v1/me/` when the user logs in, throw a warning with link/directions on how to fix.

### CODE REVIEW

**Comment by chouseknecht** [comment]
@newswangerd 

How about, if you edit it, it becomes 'unverified'. That seems easier to implement.

**Comment by chouseknecht** [comment]
@newswangerd 

Added a custom validator that prevents modifying a verified email address.

**Comment by newswangerd** [comment]
Emails should definitely be verified. 

My proposal would be as follows (based on the UI in #1072):
1. User hits the verify button and receives a notification that an email has been sent.
screen shot 2018-08-21 at 8 50 13 am

2. The preferences page begins polling the API for email updates

3. The "Verify Email" button changes to "Resend Email" until the verification confirmation is received. 

4. Once the confirmation is received, stop polling the API, mark the email as verified, and disable the "Verify Email" button.

**Comment by chouseknecht** [comment]
Removed WIP. We agreed to add email verification workflow as a separate PR.

File: galaxy/api/views/views.py [hunk 1]
```diff
@@ -168,7 +168,11 @@ def get(self, request, format=None):
         data['search'] = reverse('api:search_view')
         data['tags'] = reverse('api:tag_list')
         data['users'] = reverse('api:user_list')
+<<<<<<< HEAD
```
Comment (cutwater) [inline galaxy/api/views/views.py]: Merge conflict unresolved.

Comment (chouseknecht) [reply galaxy/api/views/views.py]: Fixed.

**Pull request merged**